### PR TITLE
#48 Modbus possibly more stable when not closing connection after every call.

### PR DESCRIPTION
**Pull request by ArdJonker**
- Just opening the connection, not closing anymore. Only when configuration changes.
- Removed function to always have one call open at a time as well.
- Reverting to defaults for modbus client timeout, retries, reconnect_delay and _max.

### CODE REVIEW

**Comment by ArdJonker** [comment]
It has run on test and production for half a day. I'll wait an other two days before merging.

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py [hunk 1]
```diff
@@ -274,20 +273,22 @@ async def initialize(self):
 
     async def initialise_charger(self, v2g_args=None):
         self.log(f"Configuring Modbus EVSE client at {c.CHARGER_HOST_URL}:{c.CHARGER_PORT}, reason: {v2g_args}")
+
+        # Remove old client if needed.
+        if self.client is not None:
+            if self.client.connected:
+                self.client.close()
+            self.client = None
```
Comment (rhpijnacker) [inline v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py:281]: Setting the client to `None` has little added value. In the next line you are assigning it a new value. I think this line can be safely removed.

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py [hunk 2]
```diff
@@ -1115,11 +1034,8 @@ async def __force_get_register(self, register: int, min: int, max: int) -> int:
 
             if result is not None:
                 try:
-                    result = int(float(result.registers[0]))
-                    if result > self.HALF_MAX_USI:
-                        # This represents a negative value.
-                        result = result - self.MAX_USI
-                    if min <= result <= max:
+                    result = self.__get_2comp(result.registers[0])
```
Comment (rhpijnacker) [inline v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py:1037]: This looks like an unrelated change. Did you mean to include this?

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py [hunk 3]
```diff
@@ -274,20 +273,22 @@ async def initialize(self):
...
         self.client = modbusClient.AsyncModbusTcpClient(
             host=c.CHARGER_HOST_URL,
             port=c.CHARGER_PORT,
-            timeout=3,
-            retries=3,
-            reconnect_delay=500,
-            reconnect_delay_max=1000,
             retry_on_empty=True,
         )
         await self.client.connect()
```
Comment (rhpijnacker) [inline v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py:288]: I would expect some error handling for the scenario when the `connect()` call fails. Or is that handled somewhere else and did I miss it?

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py (hunk 1 above)
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py:281]: Will do

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py (hunk 2 above)
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py:1037]: You are right, unrelated but relevant. Will take more care is splitting up in the future.

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py (hunk 3 above)
Comment (ArdJonker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g-liberty/modbus_evse_client.py:288]: Strangely enough the (connection-) errors occur not while connecting but while reading / writing. So that is where the error handling is now is focussed.
